protocol/x/clob/keeper/stateful_order_state.go

```diff
@@ -175,23 +175,67 @@ func (k Keeper) DeleteLongTermOrderPlacement(
 	)
 }
 
-// GetStatefulOrdersTimeSlice gets a slice of stateful order IDs that expire at `goodTilBlockTime`,
-// sorted by order ID.
-func (k Keeper) GetStatefulOrdersTimeSlice(ctx sdk.Context, goodTilBlockTime time.Time) (
+// GetStatefulOrderIdExpirations gets a slice stateful order IDs that expire at `goodTilBlockTime`,
+// sorted by order ID state key.
+func (k Keeper) GetStatefulOrderIdExpirations(ctx sdk.Context, goodTilBlockTime time.Time) (
 	orderIds []types.OrderId,
 ) {
-	store := k.getStatefulOrdersTimeSliceStore(ctx)
-	statefulOrdersTimeSliceBytes := store.Get(sdk.FormatTimeBytes(goodTilBlockTime))
+	return k.GetOrderIds(
+		ctx,
+		ctx.KVStore(k.storeKey),
+		fmt.Sprintf(types.StatefulOrdersExpirationsKeyPrefix, sdk.FormatTimeString(goodTilBlockTime)),
+	)
+}
 
-	// If there are no stateful orders that expire at this block, then return an empty slice.
-	if statefulOrdersTimeSliceBytes == nil {
-		return []types.OrderId{}
-	}
+// RemoveStatefulOrderIdExpiration removes a stateful order id expiration for an order id and
+// `goodTilBlockTime`.
+func (k Keeper) RemoveStatefulOrderIdExpiration(
+	ctx sdk.Context,
+	goodTilBlockTime time.Time,
+	orderId types.OrderId,
+) {
+	k.RemoveUnorderedOrderId(
+		ctx,
+		ctx.KVStore(k.storeKey),
+		fmt.Sprintf(types.StatefulOrdersExpirationsKeyPrefix, sdk.FormatTimeString(goodTilBlockTime)),
+		orderId,
+	)
+}
 
-	var longTermOrders types.StatefulOrderTimeSliceValue
-	k.cdc.MustUnmarshal(statefulOrdersTimeSliceBytes, &longTermOrders)
+// AddStatefulOrderIdExpiration adds a stateful order id expiration for an order id and
+// `goodTilBlockTime`.
+func (k Keeper) AddStatefulOrderIdExpiration(
+	ctx sdk.Context,
+	goodTilBlockTime time.Time,
+	orderId types.OrderId,
+) {
+	k.SetUnorderedOrderId(
+		ctx,
+		ctx.KVStore(k.storeKey),
+		fmt.Sprintf(types.StatefulOrdersExpirationsKeyPrefix, sdk.FormatTimeString(goodTilBlockTime)),
+		orderId,
+	)
+}
 
-	return longTermOrders.OrderIds
+// RemoveExpiredStatefulOrders removes the stateful order id expirations up to `blockTime` and
+// returns the removed order ids as a slice.
+func (k Keeper) RemoveExpiredStatefulOrders(ctx sdk.Context, blockTime time.Time) (
+	expiredOrderIds []types.OrderId,
+) {
+	expiredOrderIds = make([]types.OrderId, 0)
+	store := ctx.KVStore(k.storeKey)
+	start := []byte(fmt.Sprintf(types.StatefulOrdersExpirationsKeyPrefix, sdk.FormatTimeString(time.Time{})))
+	end := []byte(fmt.Sprintf(types.StatefulOrdersExpirationsKeyPrefix, sdk.FormatTimeString(blockTime)))
+	// Increment the byte representation of `/` at the end of the prefix so that blockTime is included in the iteration
```

Comment:
```suggestion
	// Increment the byte representation of `:` at the end of the prefix so that blockTime is included in the iteration
```
Do you mean this, given the key prefix format is `Exp/%s:`?